integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/datamap/CarbonCreateDataMapCommand.scala

```diff
@@ -69,8 +69,29 @@ case class CarbonCreateDataMapCommand(
     }
 
     dataMapSchema = new DataMapSchema(dataMapName, dmClassName)
-    if (mainTable != null &&
-        mainTable.isStreamingTable &&
+    if (dataMapSchema.getProviderName.equalsIgnoreCase(DataMapClassProvider.LUCENE.toString)) {
```

Comment:
This all validation code should be inside the luceneDatamapFactory init method. Please don't add any lucene specific code outside of lucene classes